patroni/postgresql.py

```diff
@@ -223,24 +223,72 @@ def sync_from_leader(self, leader):
         r = parseurl(leader.conn_url)
 
         env = self.write_pgpass(r)
-        return self.create_replica(r, env) == 0
+        ret = self.create_replica(leader, env) == 0
+        ret and self.delete_trigger_file()
+        return ret
 
     @staticmethod
     def build_connstring(conn):
-        return "host={host} port={port} user={user}".format(**conn)
+        """
+        >>> Postgresql.build_connstring({'host': '127.0.0.1', 'port': '5432'}) == 'host=127.0.0.1 port=5432 '
+        True
+        """
+        mconn = ""
+        for param, val in sorted(conn.items()):
+            mconn = mconn + "{0}={1} ".format(param, val)
+
+        return mconn
+
+    def create_replica(self, leader, env):
+        # create the replica according to the replica_method
+        # defined by the user.  this is a list, so we need to
+        # loop through all methods the user supplies
+        connstring = leader.conn_url
+        # get list of replica methods from config
+        replica_list = self.config.get('create_replica_method', 'basebackup')
```

Comment:
@feikesteenbergen : yeah, that seems like a good idea.  Less vulnerable to punctuation wierdness.  Wanna add it?